src/object-to-ast.ts

```diff
@@ -0,0 +1,48 @@
+import type babelCore from '@babel/core'
+
+module.exports = objectToAST
+
+type ObjectToASTOptions = {
+  babel: typeof babelCore
+  fileOptions: babelCore.TransformOptions
+}
+
+export default function objectToAST(
+  object: unknown,
+  {babel, fileOptions}: ObjectToASTOptions,
+) {
+  const stringified = stringify(object)
+  const variableDeclarationNode = babel.template(`var x = ${stringified}`, {
+    preserveComments: true,
+    placeholderPattern: false,
+    ...fileOptions.parserOpts,
+    sourceType: 'module',
+  })()
+
+  // @ts-expect-error "field is missing in declaration, but available at runtime?"
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+  return variableDeclarationNode.declarations[0].init as babelCore.Node
```

Comment:
This is frustrating me.  Using the [same types from `babel-plugin-codegen`](https://github.com/kentcdodds/babel-plugin-codegen/blob/9bb9dfd26a6ba55654bd493ee73b9778f8e66523/src/helpers.ts#L37-L40) it's getting an error that `declarations` doesn't exist in the returned type from `babel.template()`.

Logging it at runtime shows the field is there, so not sure where to go from here?

Reply:
In the example you linked, `typeof` seems to be used, I wonder if we can abstract `babelCore.Node` into a type, and then just cast to that type.

`Like type Node = typeof babelCore.Node`

And this line would be something like: `variableDeclarationNode.declarations[0].init as Node`

Reply:
The return type is less of an issue right now (what I have here is working), but rather the path to get to `init` doesn't exist at compile time, according to the types, but does at runtime. Chances are fixing the `declarations` will make the case unnecessary.

I'm not sure if:
1. The type is wrong and `babel.template()` _isnt_ the `@babel/core` version
   * If not, then what _is_ the correct type?
2. The type just needs narrowing to prove to TS that it is there
   * I can't find any reference to `definitions` when spelunking through the definition files, but might just be missing it?
3. The type definitions are actually wrong
   * This seems unlikely